Makefile

```diff
@@ -1,102 +1,170 @@
-	$(VENV_BIN)/pip install --upgrade -r requirements.txt
-	$(VENV_BIN)/pip install --upgrade -r docs/requirements.txt
+# Define the path to main Python pyproject.toml
+PYPROJECT_PATH := python/quantum-pecos/pyproject.toml
 
-updatereqs: upgrade-pip  ## Autogenerate requirements.txt
-	$(VENV_BIN)/pip install -U pip-tools
-	-@rm requirements.txt
-	$(VENV_BIN)/pip-compile --extra=tests --no-annotate --no-emit-index-url --output-file=requirements.txt --strip-extras pyproject.toml
+# Virtual environment
+# -------------------
 
-metadeps: upgrade-pip  ## Install extra dependencies used to develop/build this package
-	$(VENV_BIN)/pip install -U build pip-tools pre-commit wheel pytest hypothesis
+.PHONY: venv
+venv:  ## Build Python virtual environment and install requirements
+	$(PYTHONPATH) -m venv $(VENV)
+	$(MAKE) requirements
 
-# Installation
+# Requirements
 # ------------
+.PHONY: requirements
+requirements: updatereqs installreqs metadeps  ## Install/refresh Python project requirements including those needed for development
 
-install: upgrade-pip  ## Install PECOS
-	$(VENV_BIN)/pip install .
-
-install-all: upgrade-pip  ## Install PECOS with all optional dependencies
-	$(VENV_BIN)/pip install .[all]
+.PHONY: updatereqs
+updatereqs:  ## Auto update and generate requirements.txt
+	@echo "Upgrading pip..."
+	$(VENV_BIN)/python -m pip install --upgrade pip
+	$(VENV_BIN)/pip install -U pip-tools
+	-@rm python/quantum-pecos/requirements.txt
+	@echo "Using version: $(call get_version)"
+	@echo "Temporarily modifying pyproject.toml..."
+	@sed -i.bak '/pecos-rslib==/d' $(PYPROJECT_PATH)
+	$(VENV_BIN)/pip-compile --extra=tests --no-annotate --no-emit-index-url \
+		--output-file=python/quantum-pecos/requirements.txt \
+		--strip-extras $(PYPROJECT_PATH)
+	@echo "Restoring original pyproject.toml..."
+	@mv $(PYPROJECT_PATH).bak $(PYPROJECT_PATH)
+	@echo "Adding pecos-rslib back to requirements.txt..."
+	@echo "pecos-rslib==$(call get_version)" >> python/quantum-pecos/requirements.txt
+
+.PHONY: installreqs
+installreqs: upgrade-pip ## Install Python project requirements
+	@echo "Temporarily removing pecos-rslib from requirements..."
+	@grep -v "pecos-rslib" python/quantum-pecos/requirements.txt > temp_requirements.txt
+
+	@echo "Installing project requirements (excluding pecos-rslib)..."
+	$(VENV_BIN)/pip install -r temp_requirements.txt
+
+	@echo "Cleaning up temporary files..."
+	@rm temp_requirements.txt
+
+.PHONY: metadeps
+metadeps: upgrade-pip  ## Install extra dependencies used to develop/build this package
+	$(VENV_BIN)/pip install -U pip-tools pre-commit pytest hypothesis maturin
+
+# Function to extract version from pyproject.toml
+define get_version
+$(shell grep -m 1 'version\s*=' $(PYPROJECT_PATH) | sed 's/.*version\s*=\s*"\(.*\)".*/\1/')
+endef
+
+# Building development environments
+# ---------------------------------
+.PHONY: build
+build: venv ## Compile and install for development
```

Comment:
This fails for me with: 
```bash
RROR: Could not find a version that satisfies the requirement pecos-rslib==0.6.0.dev6 (from quantum-pecos) (from versions: none)
ERROR: No matching distribution found for pecos-rslib==0.6.0.dev6
```

I am assuming I need to make requirements first? I will try that and then retry `make build`

Reply:
Hmmm same error, but looking at the fuller dump, I'm very confused because it looks like the package is being installed and then not used:

```bash
📦 Including license file "/home/peter/Downloads/PECOS-feat-rust/python/pecos-rslib/LICENSE"
🍹 Building a mixed python/rust project
🔗 Found pyo3 bindings
🐍 Found CPython 3.10 at /home/peter/Downloads/rust-pecos/bin/python
📡 Using build options features from pyproject.toml
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.07s
📦 Built wheel for CPython 3.10 to /tmp/.tmpzQTl39/pecos_rslib-0.6.0.dev6-cp310-cp310-linux_x86_64.whl
✏️  Setting installed package as editable
🛠 Installed pecos-rslib-0.6.0.dev6
Looking in indexes: https://john.campora%40quantinuum.com:****@quantinuumhw.jfrog.io/artifactory/api/pypi/qtm-pypi-vc/simple
Obtaining file:///home/peter/Downloads/PECOS-feat-rust/python/quantum-pecos
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
INFO: pip is looking at multiple versions of quantum-pecos to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement pecos-rslib==0.6.0.dev6 (from quantum-pecos) (from versions: none)
ERROR: No matching distribution found for pecos-rslib==0.6.0.dev6
make: *** [Makefile:75: build] Error 1
```

Reply:
Okay, now you will have to make a .venv either via the `venv` command or by hand. If this doesn't "fix" this. Might need to take out all the $(VENV_BIN)/ and assume the user has dropped into the virtual environment